src/commands/importExport/import.ts

```diff
@@ -85,88 +42,47 @@ export const register: RegisterCommand = yargs =>
+    handler: args => importEnvironmentCli(args).catch(simplifyErrors),
   });
 
-type ImportEntitiesParams =
+type ImportEnvironmentCliParams =
   & Readonly<{
     environmentId: string;
     fileName: string;
     apiKey: string;
-    include?: ReadonlyArray<string>;
-    exclude?: ReadonlyArray<string>;
+    include?: ReadonlyArray<ImportEntityChoices>;
+    exclude?: ReadonlyArray<ImportEntityChoices>;
   }>
   & LogOptions;
 
-const importEntities = async (params: ImportEntitiesParams) => {
-  const root = new StreamZip.async({ file: params.fileName });
+const importEnvironmentCli = async (params: ImportEnvironmentCliParams) => {
   const client = createClient({
     environmentId: params.environmentId,
     apiKey: params.apiKey,
     commandName,
   });
 
-  const shouldImport = (name: string) =>
-    (!params.include || params.include.includes(name)) && !params.exclude?.includes(name);
-
-  const definitionsToImport = entityDefinitions
-    .filter(e => shouldImport(e.name) || (e.isDependentOn && shouldImport(e.isDependentOn)));
-
-  logInfo(
-    params,
-    "standard",
-    `Importing entities from ${chalk.blue(params.fileName)} into environment id ${params.environmentId}\n`,
-  );
-
-  let context = createInitialContext();
-
-  await serially(definitionsToImport.map(def => async () => {
-    logInfo(params, "standard", `Importing: ${chalk.yellow(def.displayName)}`);
-
-    try {
-      context = await root.entryData(`${def.name}.json`)
-        .then(b => b.toString("utf8"))
-        .then(def.deserializeEntities)
-        .then(e => def.importEntities(client, e, context, params, root))
-        ?? context;
-    } catch (err) {
-      logError(
-        params,
-        `Failed to import entity ${chalk.red(def.displayName)}. `,
-        JSON.stringify(err, Object.getOwnPropertyNames(err)),
-        "\nStopping import...",
-      );
-      process.exit(1);
-    }
-  }));
-
-  logInfo(params, "standard", chalk.green("All entities were successfully imported."));
+  try {
+    await importEnvironmentInternal(client, resolveParams(params));
+  } catch (e: unknown) {
+    logError(params, `${JSON.stringify(e, Object.getOwnPropertyNames(e))}\nStopping import...`);
+    process.exit(1);
+  }
 };
 
-const createInitialContext = (): ImportContext => ({
-  collectionIdsByOldIds: new Map(),
-  languageIdsByOldIds: new Map(),
-  taxonomyGroupIdsByOldIds: new Map(),
-  taxonomyTermIdsByOldIds: new Map(),
-  assetFolderIdsByOldIds: new Map(),
-  assetIdsByOldIds: new Map(),
-  oldAssetCodenamesByIds: new Map(),
-  contentTypeSnippetContextByOldIds: new Map(),
-  contentItemContextByOldIds: new Map(),
-  oldContentItemCodenamesByIds: new Map(),
-  contentTypeContextByOldIds: new Map(),
-  workflowIdsByOldIds: new Map(),
-  workflowStepsIdsWithTransitionsByOldIds: new Map(),
-  spaceIdsByOldIds: new Map(),
-});
+const resolveParams = (params: ImportEnvironmentCliParams): ImportEnvironmentParams =>
```

Comment:
What about putting this into the `includeExclude.ts` as something like `resolveIncludeExcludeFromCli`?